From aa0109eadaf4b8bab61fc5851e8d895250df597c Mon Sep 17 00:00:00 2001
From: Dan Nica <dan.nica@nxp.com>
Date: Fri, 22 Nov 2019 11:45:10 +0200
Subject: [PATCH 025/269] s32g: Add missing includes in platform.mk

Issue: ALB-3895
Upstream-Status: Pending 

Signed-off-by: Dan Nica <dan.nica@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/s32g/platform.mk | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/plat/s32g/platform.mk b/plat/s32g/platform.mk
index a694ac048..aa3cce5d6 100644
--- a/plat/s32g/platform.mk
+++ b/plat/s32g/platform.mk
@@ -8,13 +8,15 @@ include lib/xlat_tables_v2/xlat_tables.mk
 
 PLAT_INCLUDES		+= -Iplat/s32g/include \
 			   -Iinclude/common/tbbr \
+			   -Iinclude/plat/common \
 			   -Iinclude/plat/arm/common \
 			   -Iinclude/plat/arm/soc/common \
+			   -Iinclude/lib \
+			   -Iinclude/drivers \
 			   -Iinclude/lib/psci
 PLAT_BL_COMMON_SOURCES	+= plat/s32g/s32g_helpers.S \
 			   plat/s32g/include/plat_macros.S \
-			   plat/s32g/s32g_xrdc.c \
-			   drivers/console/aarch64/console.S
+			   plat/s32g/s32g_xrdc.c
 PLAT_BL_COMMON_SOURCES	+= ${XLAT_TABLES_LIB_SRCS}
 
 BL31_SOURCES		+= plat/s32g/s32g275_bl31.c \
-- 
2.17.1

