From 957b32af36911848452d09e2b83ea6b6425cf001 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Date: Thu, 21 Jan 2021 15:18:31 +0200
Subject: [PATCH 46/78] rtc-s32gen1: Implement set_time callback

Issue: ALB-5823
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
---
 drivers/rtc/rtc-s32gen1.c | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/drivers/rtc/rtc-s32gen1.c b/drivers/rtc/rtc-s32gen1.c
index e3f8a1bc8260..be8a0f9a2220 100644
--- a/drivers/rtc/rtc-s32gen1.c
+++ b/drivers/rtc/rtc-s32gen1.c
@@ -243,8 +243,24 @@ static int s32gen1_rtc_set_alarm(struct device *dev, struct rtc_wkalrm *alrm)
 	return err;
 }
 
+static int s32gen1_rtc_set_time(struct device *dev, struct rtc_time *time)
+{
+	struct rtc_s32gen1_priv *priv = dev_get_drvdata(dev);
+	u32 rtccnt = ioread32(priv->rtc_base + RTCCNT_OFFSET);
+
+	if (!time)
+		return -EINVAL;
+
+	priv->base.rollovers = priv->rollovers;
+	priv->base.cycles = rtccnt;
+	rtc_tm_to_time(time, &priv->base.sec);
+
+	return 0;
+}
+
 static const struct rtc_class_ops s32gen1_rtc_ops = {
 	.read_time = s32gen1_rtc_read_time,
+	.set_time = s32gen1_rtc_set_time,
 	.read_alarm = s32gen1_rtc_read_alarm,
 	.set_alarm = s32gen1_rtc_set_alarm,
 	.alarm_irq_enable = s32gen1_alarm_irq_enable,
-- 
2.25.1

