From b5673a37c730968c072f56b344802b902390c66a Mon Sep 17 00:00:00 2001
From: Ondrej Spacek <ondrej.spacek@nxp.com>
Date: Mon, 1 Feb 2021 12:34:43 +0100
Subject: [PATCH 55/78] dts: s32-gen1: Add timestamp clock muxes to kernel.

Issue: ALB-6407
Signed-off-by: Ondrej Spacek <ondrej.spacek@nxp.com>
---
 drivers/clk/s32/s32-gen1/clk.c | 25 +++++++++++++++++++++++++
 1 file changed, 25 insertions(+)

diff --git a/drivers/clk/s32/s32-gen1/clk.c b/drivers/clk/s32/s32-gen1/clk.c
index 84f5b23a3c2a..15ca450e8057 100644
--- a/drivers/clk/s32/s32-gen1/clk.c
+++ b/drivers/clk/s32/s32-gen1/clk.c
@@ -100,6 +100,12 @@ static u32 gmac_0_tx_mux_idx[] = {
 	MC_CGM_MUXn_CSC_SEL_SERDES_0_LANE_0_TX_CLK,
 };
 
+PNAME(gmac_0_ts_sels) = {"firc", "periphpll_phi4", "periphpll_phi5", };
+static u32 gmac_0_ts_mux_idx[] = {
+	MC_CGM_MUXn_CSC_SEL_FIRC, MC_CGM_MUXn_CSC_SEL_PERIPH_PLL_PHI4,
+	MC_CGM_MUXn_CSC_SEL_PERIPH_PLL_PHI5,
+};
+
 PNAME(gmac_1_tx_sels) = {"firc", "periphpll_phi5", "serdes_1_lane_0", };
 static u32 gmac_1_tx_mux_idx[] = {
 	MC_CGM_MUXn_CSC_SEL_FIRC, MC_CGM_MUXn_CSC_SEL_PERIPH_PLL_PHI5,
@@ -660,6 +666,15 @@ void __init s32gen1_clocks_init(struct device_node *clocking_node)
 				clk_modules.mc_cgm0_base, 10, &s32gen1_lock);
 	set_plat_clk(S32GEN1_CLK_GMAC_0_TX, c);
 
+	c = s32gen1_clk_cgm_mux("gmac_0_ts_sel", clk_modules.mc_cgm0_base,  9,
+				gmac_0_ts_sels, ARRAY_SIZE(gmac_0_ts_sels),
+				gmac_0_ts_mux_idx, &s32gen1_lock);
+	set_plat_clk(S32GEN1_CLK_GMAC_0_TS_SEL, c);
+
+	c = s32gen1_clk_cgm_div("gmac_0_ts", "gmac_0_ts_sel",
+				clk_modules.mc_cgm0_base, 9, &s32gen1_lock);
+	set_plat_clk(S32GEN1_CLK_GMAC_0_TS, c);
+
 	/* Add the clocks to provider list */
 	plat_clks.plat_clks.clks = clk;
 	plat_clks.plat_clks.clk_num = ARRAY_SIZE(clk);
@@ -701,10 +716,14 @@ void s32gen1_scmi_clocks_init(void)
 		get_plat_clk(S32GEN1_CLK_GMAC_0_RX);
 	scmi_clk[S32GEN1_SCMI_CLK_GMAC0_TX_SGMII] =
 		get_plat_clk(S32GEN1_CLK_GMAC_0_TX);
+	scmi_clk[S32GEN1_SCMI_CLK_GMAC0_TS_SGMII] =
+		get_plat_clk(S32GEN1_CLK_GMAC_0_TS);
 	scmi_clk[S32GEN1_SCMI_CLK_GMAC0_RX_RGMII] =
 		get_plat_clk(S32GEN1_CLK_GMAC_0_RX);
 	scmi_clk[S32GEN1_SCMI_CLK_GMAC0_TX_RGMII] =
 		get_plat_clk(S32GEN1_CLK_GMAC_0_TX);
+	scmi_clk[S32GEN1_SCMI_CLK_GMAC0_TS_RGMII] =
+		get_plat_clk(S32GEN1_CLK_GMAC_0_TS);
 	scmi_clk[S32GEN1_SCMI_CLK_GMAC0_RX_RMII] =
 		get_plat_clk(S32GEN1_CLK_GMAC_0_RX);
 	scmi_clk[S32GEN1_SCMI_CLK_GMAC0_TX_RMII] =
@@ -713,6 +732,8 @@ void s32gen1_scmi_clocks_init(void)
 		get_plat_clk(S32GEN1_CLK_GMAC_0_RX);
 	scmi_clk[S32GEN1_SCMI_CLK_GMAC0_TX_MII] =
 		get_plat_clk(S32GEN1_CLK_GMAC_0_TX);
+	scmi_clk[S32GEN1_SCMI_CLK_GMAC0_TS_MII] =
+		get_plat_clk(S32GEN1_CLK_GMAC_0_TS);
 	scmi_clk[S32GEN1_SCMI_CLK_GMAC0_AXI] =
 		get_plat_clk(S32GEN1_CLK_XBAR);
 
@@ -822,6 +843,10 @@ static void s32g274a_scmi_clocks_init(void)
 	scmi_clk[S32G274A_SCMI_CLK_PFE_PE] =
 		get_plat_clk(S32GEN1_CLK_PFE_PE);
 
+	/* PFE timestamp clock */
+	scmi_clk[S32G274A_SCMI_CLK_PFE_TS] =
+		get_plat_clk(S32GEN1_CLK_GMAC_0_TS);
+
 	/* PFE 0 */
 	scmi_clk[S32G274A_SCMI_CLK_PFE0_RX_SGMII] =
 		get_plat_clk(S32GEN1_CLK_PFE_EMAC_0_RX);
-- 
2.25.1

