From df86fa59483aeac753586c6b7f50d3e38a76297b Mon Sep 17 00:00:00 2001
From: Vlad Pelin <vlad.pelin@nxp.com>
Date: Thu, 18 Nov 2021 19:35:49 +0200
Subject: [PATCH 2/2] pkcs: update makefile for hse fw 1.0.0

Issue: ALB-8211
Upstream-Status: Pending 

Signed-off-by: Vlad Pelin <vlad.pelin@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 Makefile | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/Makefile b/Makefile
index e6c45af..da5d663 100644
--- a/Makefile
+++ b/Makefile
@@ -2,11 +2,11 @@
 # Copyright 2021 NXP
 #
 
-PLATFORM ?= S32G274
+PLATFORM ?= S32G2
 FWTYPE ?= 0
-FWMAJOR ?= 0
-FWMINOR ?= 9
-FWPATCH ?= 2
+FWMAJOR ?= 1
+FWMINOR ?= 0
+FWPATCH ?= 0
 
 # Skip prerequisites check when invoking make clean
 ifeq ($(filter clean,$(MAKECMDGOALS)),$(strip $(MAKECMDGOALS)))
@@ -23,7 +23,7 @@ ifeq (,$(CROSS_COMPILE))
 endif
 
 ifeq (,$(HSE_FWDIR))
-    HSE_FWDIR ?= $(HOME)/HSE_$(PLATFORM)_$(FWTYPE)_$(FWMAJOR)_$(FWMINOR)_$(FWPATCH)
+    HSE_FWDIR ?= $(HOME)/HSE_FW_$(PLATFORM)_$(FWTYPE)_$(FWMAJOR)_$(FWMINOR)_$(FWPATCH)
     $(warning Path to HSE firmware package not defined, using default $(HSE_FWDIR))
 endif
 
-- 
2.17.1

