From 37deba24cd0e287b1300d5621472eb6155e9749e Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Wed, 27 Oct 2021 12:20:16 +0300
Subject: [PATCH 04/19] s32g3: Correct Linux device tree name

Issue: ALB-8016
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/configs/s32g398a.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/configs/s32g398a.h b/include/configs/s32g398a.h
index 468d9d39e8..d95658093f 100644
--- a/include/configs/s32g398a.h
+++ b/include/configs/s32g398a.h
@@ -13,7 +13,7 @@
 #include <configs/s32-gen1.h>
 
 #if defined(CONFIG_TARGET_S32G3XXAEVB)
-#define FDT_FILE			fsl-s32g3xxaevb.dtb
+#define FDT_FILE			fsl-s32g3xxa-evb.dtb
 
 #if defined(CONFIG_USB)
 #define CONFIG_USB_EHCI_MX6
-- 
2.17.1

