From 2a8e35c6ec773524c3fda1847e5d6209ce82f6d2 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Tue, 5 Oct 2021 06:24:52 +0300
Subject: [PATCH 06/14] s32-gen1/eth: Fix compilation warning

Issue: ALB-7919
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 board/freescale/s32-gen1/eth.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/board/freescale/s32-gen1/eth.c b/board/freescale/s32-gen1/eth.c
index 794a1c481c..b73b92f73e 100644
--- a/board/freescale/s32-gen1/eth.c
+++ b/board/freescale/s32-gen1/eth.c
@@ -28,6 +28,7 @@
 #include <dm/pinctrl.h>
 #include <hwconfig.h>
 
+#if CONFIG_IS_ENABLED(DWC_ETH_QOS_S32CC) || CONFIG_IS_ENABLED(FSL_PFENG)
 static void ft_update_eth_addr_by_name(const char *name, const u8 idx,
 				       void *fdt, int nodeoff)
 {
@@ -38,6 +39,7 @@ static void ft_update_eth_addr_by_name(const char *name, const u8 idx,
 		printf("DT: %s%i set to %pM\n", name, idx, ea);
 	}
 }
+#endif
 
 #if CONFIG_IS_ENABLED(FSL_PFENG)
 __maybe_unused static bool intf_is_xmii(u32 intf)
-- 
2.17.1

