From 797a7828f9ae229f551d1889dfc4d13ffb4932fa Mon Sep 17 00:00:00 2001
From: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Date: Mon, 28 Feb 2022 16:41:59 +0200
Subject: [PATCH 20/51] s32: timer: add timer_get_boot_us

Issue: ALB-8547
Upstream-Status: Pending 

Signed-off-by: Bogdan-Gabriel Roman <bogdan-gabriel.roman@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/timer/s32_timer.c | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/drivers/timer/s32_timer.c b/drivers/timer/s32_timer.c
index 25bf6eadaf..602dbbebfe 100644
--- a/drivers/timer/s32_timer.c
+++ b/drivers/timer/s32_timer.c
@@ -52,6 +52,25 @@ static int s32_timer_get_count(struct udevice *dev, u64 *count)
 	return 0;
 }
 
+ulong timer_get_boot_us(void)
+{
+	u64 ticks = 0, us;
+	u32 rate;
+	int ret;
+
+	ret = dm_timer_init();
+
+	if (ret)
+		return 0;
+
+	/* The timer is available */
+	rate = timer_get_rate(gd->timer);
+	timer_get_count(gd->timer, &ticks);
+
+	us = (ticks * 1000000) / rate;
+	return us;
+}
+
 static int s32_timer_probe(struct udevice *dev)
 {
 	struct s32_timer_priv *priv = dev_get_priv(dev);
-- 
2.17.1

