From 6fea6bb02d8ae9fe054edfb544c0e2168f563757 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Mon, 7 Feb 2022 16:19:34 +0200
Subject: [PATCH 40/51] s32: Use booti to boot Linux

Issue: ALB-8483
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/configs/s32.h | 10 ----------
 1 file changed, 10 deletions(-)

diff --git a/include/configs/s32.h b/include/configs/s32.h
index 45f0d08a49..1eaf835934 100644
--- a/include/configs/s32.h
+++ b/include/configs/s32.h
@@ -109,18 +109,8 @@
 
 #ifdef CONFIG_CMD_BOOTI
 
-/*
- * Enable CONFIG_USE_BOOTI if the u-boot enviroment variables
- * specific for boot method have to be defined for booti by default.
- */
-#define CONFIG_USE_BOOTI
-#ifdef CONFIG_USE_BOOTI
 #define IMAGE_NAME Image
 #define BOOT_MTD booti
-#else
-#define IMAGE_NAME uImage
-#define BOOT_MTD bootm
-#endif
 
 #endif
 
-- 
2.17.1

