From 9269054e799bfed3a2f25da0bc14e98640e1ae9e Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Mon, 7 Feb 2022 17:18:16 +0200
Subject: [PATCH 47/51] s32-gen1: Remove Linflex defines leftovers

Issue: ALB-8484
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/configs/s32-gen1.h | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/include/configs/s32-gen1.h b/include/configs/s32-gen1.h
index efe0742d8c..905d7c458d 100644
--- a/include/configs/s32-gen1.h
+++ b/include/configs/s32-gen1.h
@@ -28,14 +28,6 @@
 #define CONFIG_USE_IRQ
 #endif
 
-#if CONFIG_FSL_LINFLEX_MODULE == 0
-#define LINFLEXUART_BASE	LINFLEXD0_BASE_ADDR
-#elif CONFIG_FSL_LINFLEX_MODULE == 1
-#define LINFLEXUART_BASE	LINFLEXD1_BASE_ADDR
-#else
-#define LINFLEXUART_BASE	LINFLEXD2_BASE_ADDR
-#endif
-
 /* memory mapped external flash */
 #define CONFIG_SYS_FSL_FLASH0_BASE      0x0
 #define CONFIG_SYS_FSL_FLASH0_SIZE      0x20000000
-- 
2.17.1

