From 2d95f49b5e1284b978435932165f2431f883852e Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Mon, 7 Feb 2022 17:21:58 +0200
Subject: [PATCH 48/51] s32-gen1: Remove FEC specific macros

Issue: ALB-8484
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/configs/s32-gen1.h | 2 --
 1 file changed, 2 deletions(-)

diff --git a/include/configs/s32-gen1.h b/include/configs/s32-gen1.h
index 905d7c458d..91a62737f4 100644
--- a/include/configs/s32-gen1.h
+++ b/include/configs/s32-gen1.h
@@ -40,8 +40,6 @@
 /* we include this file here because it depends on the above definitions */
 #include <configs/s32.h>
 
-#define IMX_FEC_BASE            ENET0_BASE_ADDR
-
 #define CONFIG_SYS_TEXT_OFFSET      0x00020000
 
 #define S32_SRAM_BASE		0x34000000
-- 
2.17.1

