From 0b50e0dba1780edf43552afe14e70c6a80cf4c51 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Wed, 27 Oct 2021 11:43:27 +0300
Subject: [PATCH 03/14] s32g3: Correct the way SRAMC addresses are calculated

Issue: ALB-8016
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32g/s32g3/s32g3_sramc.c | 9 +++++++--
 1 file changed, 7 insertions(+), 2 deletions(-)

diff --git a/plat/nxp/s32/s32g/s32g3/s32g3_sramc.c b/plat/nxp/s32/s32g/s32g3/s32g3_sramc.c
index c70996145..7634cf91d 100644
--- a/plat/nxp/s32/s32g/s32g3/s32g3_sramc.c
+++ b/plat/nxp/s32/s32g/s32g3/s32g3_sramc.c
@@ -51,8 +51,13 @@ uintptr_t a53_to_sramc_addr(uintptr_t addr)
 {
 	addr -= S32G_SRAM_BASE;
 
-	/* mem_addr[16:0] = {bus_addr[23:10], bus_addr[5:4]} */
-	addr = ((addr >> 10) << 2) | ((addr >> 4) & 0x3);
+	/**
+	 * mem_addr[16:0] = { (bus_addr[24:20] modulo 5),
+	 *                    bus_addr[19:8], bus_addr[5:4]};
+	 */
+	addr = ((addr & 0x30) >> 4) |
+	    (((addr & 0xFFF00) >> 8) << 2) |
+	    (((addr & 0x1F00000) >> 20) % 5) << 14;
 
 	return addr;
 }
-- 
2.17.1

