From 5ab3b54355bb7e181a9ae4495f0ea3321fa84249 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Thu, 3 Feb 2022 14:22:14 +0200
Subject: [PATCH 21/43] drivers: crypto/hse: Add config for HSE firmware file

Issue: ALB-8469
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/crypto/fsl/Kconfig | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/crypto/fsl/Kconfig b/drivers/crypto/fsl/Kconfig
index b6d34f3d27..e8f702065a 100644
--- a/drivers/crypto/fsl/Kconfig
+++ b/drivers/crypto/fsl/Kconfig
@@ -51,3 +51,11 @@ config HSE_SECBOOT
 	depends on S32_GEN1
 	help
 		Enable HSE-provided Advanced Secure Boot
+
+config HSE_FW_FILE
+	string "HSE Firmware file"
+	default "s32gen1_hse_fw.bin"
+	depends on HSE_SECBOOT
+	help
+		File to be used as firmware for NXP HSE
+		(Hardware Security Engine) accelerator.
-- 
2.17.1

