From b522bff213e9fcd8170da6ef57633833d6287029 Mon Sep 17 00:00:00 2001
From: Dan Nica <dan.nica@nxp.com>
Date: Wed, 9 Feb 2022 10:43:30 +0200
Subject: [PATCH 42/43] s32: Fix 'loadfdt' call to 'fdt_override'

Issue: ALB-8428
Upstream-Status: Pending 

Signed-off-by: Dan Nica <dan.nica@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/configs/s32.h | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/include/configs/s32.h b/include/configs/s32.h
index 97918a758d..55621ed1e7 100644
--- a/include/configs/s32.h
+++ b/include/configs/s32.h
@@ -257,6 +257,10 @@
 		"fdt rm /usdhc no-1-8-v; " \
 		"fdt resize; \0" \
 
+#define FDT_OVERRIDE \
+	"fdt_override=" \
+		" \0"
+
 #define CONFIG_FLASHBOOT_RAMDISK " ${ramdisk_addr} "
 
 #ifdef CONFIG_XEN_SUPPORT
@@ -334,8 +338,9 @@
 	"loadimage=fatload mmc ${mmcdev}:${mmcpart} ${loadaddr} ${image}\0" \
 	"loadramdisk=fatload mmc ${mmcdev}:${mmcpart} ${ramdisk_addr} ${ramdisk}\0" \
 	FDT_ENABLE_HS400ES \
+	FDT_OVERRIDE \
 	"loadfdt=fatload mmc ${mmcdev}:${mmcpart} ${fdt_addr} ${fdt_file}; " \
-		 "${fdt_override}\0" \
+		 "run fdt_override;\0" \
 	"jtagboot=echo Booting using jtag...; " \
 		"${boot_mtd} ${loadaddr} ${ramdisk_addr} ${fdt_addr}\0" \
 	"jtagsdboot=echo Booting loading Linux with ramdisk from SD...; " \
-- 
2.17.1

