From 1ea6a220c175b00e2e30229a143ab744d5ad0002 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
Date: Tue, 24 Nov 2020 10:27:31 +0200
Subject: [PATCH 10/76] drivers: clk: s32g274a: Add LLCE clocks

Issue: ALB-5829
Signed-off-by: Ghennadi Procopciuc <Ghennadi.Procopciuc@nxp.com>
---
 drivers/clk/s32/s32g274a_scmi_ids.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/clk/s32/s32g274a_scmi_ids.c b/drivers/clk/s32/s32g274a_scmi_ids.c
index cfc2a7c905..6c479efab3 100644
--- a/drivers/clk/s32/s32g274a_scmi_ids.c
+++ b/drivers/clk/s32/s32g274a_scmi_ids.c
@@ -33,6 +33,9 @@ static u32 s32g274a_scmi_ids[] = {
 	[INDEX(S32G274A_SCMI_CLK_PFE2_TX_SGMII)] = S32GEN1_SCMI_COMPLEX_CLK,
 	[INDEX(S32G274A_SCMI_CLK_PFE2_RX_RGMII)] = S32GEN1_SCMI_COMPLEX_CLK,
 	[INDEX(S32G274A_SCMI_CLK_PFE2_TX_RGMII)] = S32GEN1_SCMI_COMPLEX_CLK,
+	/* LLCE */
+	[INDEX(S32G274A_SCMI_CLK_LLCE_SYS)] = S32GEN1_CLK_XBAR_DIV2,
+	[INDEX(S32G274A_SCMI_CLK_LLCE_CAN_PE)] = S32GEN1_CLK_CAN_PE,
 };
 
 static int s32g_compound2clkid(u32 scmi_clk_id, u32 *clk_id)
-- 
2.17.1

