From 3706e214e1efc29b4be0e49cb2176d09cbd836d8 Mon Sep 17 00:00:00 2001
From: Larisa Grigore <larisa.grigore@nxp.com>
Date: Tue, 26 Jan 2021 12:16:01 +0200
Subject: [PATCH 50/76] s32g_emu: Do not select MMC_HS400_SUPPORT

Issue: ALB-6137
Signed-off-by: Larisa Grigore <larisa.grigore@nxp.com>
---
 board/freescale/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/board/freescale/Kconfig b/board/freescale/Kconfig
index 5a56bd43df..e84ed78b65 100644
--- a/board/freescale/Kconfig
+++ b/board/freescale/Kconfig
@@ -17,7 +17,7 @@ config S32_GEN1
 	select SCMI_FIRMWARE
 	select ARCH_MISC_INIT
 	select S32_GPIO
-	select MMC_HS400_SUPPORT
+	select MMC_HS400_SUPPORT if !TARGET_TYPE_S32GEN1_EMULATOR
 	help
 	  Platform Selection
 
-- 
2.17.1

