From 858627e8b658113ccd1a588f8d9e403047e2c08c Mon Sep 17 00:00:00 2001
From: Bogdan Folea <bogdan.folea@nxp.com>
Date: Thu, 13 May 2021 18:33:35 +0300
Subject: [PATCH 41/42] config: s32: remove DDR size limitation

Upstream-Status: Pending

Issue: ALB-4313
Signed-off-by: Bogdan Folea <bogdan.folea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 include/configs/s32.h | 8 --------
 1 file changed, 8 deletions(-)

diff --git a/include/configs/s32.h b/include/configs/s32.h
index e33435dd69..27a7185f16 100644
--- a/include/configs/s32.h
+++ b/include/configs/s32.h
@@ -405,13 +405,6 @@
 #define XEN_EXTRA_ENV_SETTINGS  ""
 #endif
 
-#ifdef CONFIG_S32_GEN1
-/* Limit DDR0 to 1.5 GB due to HSE driver limitation */
-#define DDR_LIMIT0 "ddr_limit0=0xE0000000;\0"
-#else
-#define DDR_LIMIT0 ""
-#endif
-
 #define INITRD_HIGH_DEFAULT 0xffffffff
 
 /* Leave room for TF-A & OPTEE */
@@ -424,7 +417,6 @@
 #define CONFIG_EXTRA_ENV_SETTINGS \
 	CONFIG_BOARD_EXTRA_ENV_SETTINGS  \
 	CONFIG_DCU_EXTRA_ENV_SETTINGS \
-	DDR_LIMIT0 \
 	"ipaddr=" S32_DEFAULT_IP \
 	ENV_FDTCONTROLADDR \
 	"serverip=10.0.0.1\0" \
-- 
2.25.1

