From 8bcd7eda7be243d654ab0b9bc0ccf8b304a1258c Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Fri, 22 Oct 2021 08:10:04 +0300
Subject: [PATCH 09/16] s32g: emu: Skip BL31 & BL33 loading

Issue: ALB-7712
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32g/s32g_bl2_el3.c | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/plat/nxp/s32/s32g/s32g_bl2_el3.c b/plat/nxp/s32/s32g/s32g_bl2_el3.c
index 096d823d6..9b4551143 100644
--- a/plat/nxp/s32/s32g/s32g_bl2_el3.c
+++ b/plat/nxp/s32/s32g/s32g_bl2_el3.c
@@ -618,6 +618,22 @@ void s32g_el3_mmu_fixup(void)
 	enable_mmu_el3(0);
 }
 
+#if S32G_EMU == 1
+static void skip_emu_images(bl_mem_params_node_t *params, size_t size)
+{
+	unsigned int image_id;
+	size_t i;
+
+	for (i = 0; i < size; i++) {
+		image_id = params[i].image_id;
+
+		if (image_id == BL31_IMAGE_ID || image_id == BL33_IMAGE_ID)
+			params[i].image_info.h.attr |=
+			    IMAGE_ATTRIB_SKIP_LOADING;
+	}
+}
+#endif
+
 void bl2_el3_early_platform_setup(u_register_t arg0, u_register_t arg1,
 				  u_register_t arg2, u_register_t arg3)
 {
@@ -646,6 +662,10 @@ void bl2_el3_early_platform_setup(u_register_t arg0, u_register_t arg1,
 	add_bl33_img_to_mem_params_descs(params, &index);
 	add_invalid_img_to_mem_params_descs(params, &index);
 
+#if S32G_EMU == 1
+	skip_emu_images(params, index);
+#endif
+
 	bl_mem_params_desc_num = index;
 }
 
-- 
2.17.1

