From 29d39a3b6b7570af7a2396273b5cb1df99ba28e8 Mon Sep 17 00:00:00 2001
From: Ciprian Costea <ciprianmarian.costea@nxp.com>
Date: Mon, 7 Feb 2022 17:14:05 +0200
Subject: [PATCH 49/50] fdt: Remove scmi U-boot fdt fixup

Issue: ALB-8356
Upstream-Status: Pending 

Signed-off-by: Ciprian Costea <ciprianmarian.costea@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32_bl2_el3.c | 107 -------------------------------------
 1 file changed, 107 deletions(-)

diff --git a/plat/nxp/s32/s32_bl2_el3.c b/plat/nxp/s32/s32_bl2_el3.c
index b0673a0d5..f124e20bd 100644
--- a/plat/nxp/s32/s32_bl2_el3.c
+++ b/plat/nxp/s32/s32_bl2_el3.c
@@ -361,109 +361,6 @@ static int ft_fixup_ddr_errata(void *blob)
 }
 #endif
 
-static int disable_clk_node(void *blob, uint32_t *phandle)
-{
-	const char *clk_path;
-	int nodeoff, ret;
-
-	clk_path = fdt_get_alias(blob, "clks");
-	if (!clk_path) {
-		ERROR("Failed to get the path of 'clks' alias\n");
-		return -EIO;
-	}
-
-	nodeoff = fdt_path_offset(blob, clk_path);
-	if (nodeoff < 0) {
-		ERROR("Failed to get offset of '%s' node\n", clk_path);
-		return nodeoff;
-	}
-
-	*phandle = fdt_get_phandle(blob, nodeoff);
-	if (*phandle == 0) {
-		ERROR("Failed to get phandle of '%s' node\n", clk_path);
-		return *phandle;
-	}
-
-	ret = fdt_setprop_string(blob, nodeoff, "status", "disabled");
-	if (ret) {
-		ERROR("Failed to disable '%s' node\n", clk_path);
-		return ret;
-	}
-
-	ret = fdt_delprop(blob, nodeoff, "phandle");
-	if (ret) {
-		ERROR("Failed to remove phandle property of '%s' node\n",
-		       clk_path);
-		return ret;
-	}
-
-	return 0;
-}
-
-static int enable_scmi_clk_node(void *blob, uint32_t phandle)
-{
-	int nodeoff, ret;
-
-	nodeoff = fdt_path_offset(blob, "/firmware/scmi/protocol@14");
-	if (nodeoff < 0) {
-		ERROR("Failed to get offset of '/firmware/scmi/protocol@14' node\n");
-		return nodeoff;
-	}
-
-	ret = fdt_setprop_cell(blob, nodeoff, "phandle", phandle);
-	if (ret) {
-		ERROR("Failed to set phandle property of '/firmware/scmi/protocol@14' node\n");
-		return ret;
-	}
-
-	ret = fdt_setprop_string(blob, nodeoff, "status", "okay");
-	if (ret) {
-		ERROR("Failed to enable '/firmware/scmi/protocol@14' node\n");
-		return ret;
-	}
-
-	return 0;
-}
-
-static int enable_scmi_mbox(void *blob)
-{
-	int nodeoff, ret;
-
-	nodeoff = fdt_node_offset_by_compatible(blob, -1, "arm,scmi-smc");
-	if (nodeoff < 0) {
-		ERROR("Failed to get offset of 'arm,scmi-smc' node\n");
-		return nodeoff;
-	}
-
-	ret = fdt_setprop_string(blob, nodeoff, "status", "okay");
-	if (ret) {
-		ERROR("Failed to enable 'arm,scmi-smc' node\n");
-		return nodeoff;
-	}
-
-	return 0;
-}
-
-static int ft_fixup_scmi_clks(void *blob)
-{
-	uint32_t phandle;
-	int ret;
-
-	ret = disable_clk_node(blob, &phandle);
-	if (ret)
-		return ret;
-
-	ret = enable_scmi_clk_node(blob, phandle);
-	if (ret)
-		return ret;
-
-	ret = enable_scmi_mbox(blob);
-	if (ret)
-		return ret;
-
-	return 0;
-}
-
 static int ft_fixup_resmem_node(void *blob)
 {
 	int ret;
@@ -545,10 +442,6 @@ static int ft_fixups(void *blob)
 	size += S32_FDT_UPDATES_SPACE;
 	fdt_set_totalsize(blob, size);
 
-	ret = ft_fixup_scmi_clks(blob);
-	if (ret)
-		goto out;
-
 #if (ERRATA_S32_050543 == 1 && S32G_EMU == 0)
 	ret = ft_fixup_ddr_errata(blob);
 	if (ret)
-- 
2.17.1

