From ac7ea2975ec81f80c91a882669e87d6a042d4e13 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Mon, 13 Dec 2021 07:31:44 +0200
Subject: [PATCH 11/14] s32g: sramc: Don't substract SRAM base from an SRAM
 offset

Issue: ALB-8270
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 plat/nxp/s32/s32g/s32g2/s32g2_sramc.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/plat/nxp/s32/s32g/s32g2/s32g2_sramc.c b/plat/nxp/s32/s32g/s32g2/s32g2_sramc.c
index df865d4ba..85032eb53 100644
--- a/plat/nxp/s32/s32g/s32g2/s32g2_sramc.c
+++ b/plat/nxp/s32/s32g/s32g2/s32g2_sramc.c
@@ -14,8 +14,6 @@
 
 static uintptr_t a53_to_sramc_offset(uintptr_t addr)
 {
-	addr -= S32G_SRAM_BASE;
-
 	/* mem_addr[16:0] = {bus_addr[23:9], bus_addr[5:4]} */
 	addr = ((addr >> 9) << 2) | ((addr >> 4) & 0x3);
 
-- 
2.17.1

